odo tries to run OpenShift tests in Kubernetes environment?

/kind bug

<!--

Welcome! - We kindly ask you to:

  1. Fill out the issue template below 
  2. Use the Google group if you have a question rather than a bug or feature request.

The group is at: https://groups.google.com/forum/#!forum/odo-users

Thanks for understanding, and for contributing to the project!

-->

## What versions of software are you using?

**Operating System:** Linux

**Output of `odo version`:** latest master

## How did you run odo exactly?

```sh
export KUBERNETES=true
ginkgo -focus="odo preference and config command tests" tests/integration
```

## Actual behavior

It tries to run the `oc login` and `oc logout` commands...


## Expected behavior

To not run ANY openshift commands.

## Any logs, error output, etc?

```sh
Running Suite: Integration Suite
================================
Random Seed: [1m1643732273[0m
Will run [1m14[0m of [1m48[0m specs

[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m[36mS[0m
[90m------------------------------[0m
[32m• [SLOW TEST:6.168 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  check that help works
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:30[0m
    should display help info
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:31[0m
[90m------------------------------[0m
[32m• [SLOW TEST:6.179 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  when running help for preference command
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:40[0m
    should display the help
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:41[0m
[90m------------------------------[0m
[32m• [SLOW TEST:6.097 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  when running help for config command
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:47[0m
    should display the help
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:48[0m
[90m------------------------------[0m
[32m• [SLOW TEST:6.506 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  when running odo config view
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:54[0m
    should pass if devfile is present in current dir
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:55[0m
[90m------------------------------[0m
[32m• [SLOW TEST:6.071 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  when running odo config view
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:54[0m
    should fail if devfile not present in current dir
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:63[0m
[90m------------------------------[0m
[32m• [SLOW TEST:6.587 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  When viewing global config
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:69[0m
    should get the default global config keys
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:81[0m
[90m------------------------------[0m
[32m• [SLOW TEST:8.496 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  When configuring global config values
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:92[0m
    should successfully updated
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:105[0m
[90m------------------------------[0m
[32m• [SLOW TEST:6.469 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  When configuring global config values
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:92[0m
    should unsuccessfully update
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:123[0m
[90m------------------------------[0m
[32m• [SLOW TEST:5.998 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  When configuring global config values
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:92[0m
    should show json output
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:132[0m
[90m------------------------------[0m
[91m[1m• Failure in Spec Setup (BeforeEach) [13.776 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  when viewing local config without logging into the OpenShift cluster
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:141[0m
    [91m[1mwhen user is working with a devfile component [BeforeEach][0m
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:154[0m
      should set, list and delete config successfully
      [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:155[0m

      [91mNo future change is possible.  Bailing out early after 0.143s.
      Expected
          <int>: 1
      to match exit code:
          <int>: 0[0m

      /home/wikus/syncthing/files/dev/go/src/odo/tests/helper/helper_oc.go:214
[90m------------------------------[0m
Created dir: /tmp/2747527892
Created dir: /tmp/2270213049
Setting KUBECONFIG=/tmp/2270213049/config
Creating a new project: cmd-pref-config-test155xnv
Running kubectl with args [kubectl create namespace cmd-pref-config-test155xnv]
[kubectl] namespace/cmd-pref-config-test155xnv created
Running kubectl with args [kubectl config set-context --current --namespace cmd-pref-config-test155xnv]
[kubectl] Context "k0s-cluster" modified.
Running kubectl with args [kubectl get namespaces]
[kubectl] NAME                         STATUS   AGE
[kubectl] cert-manager                 Active   29d
[kubectl] cmd-pref-config-test155xnv   Active   0s
[kubectl] default                      Active   44d
[kubectl] ingress-nginx                Active   41d
[kubectl] kube-node-lease              Active   44d
[kubectl] kube-public                  Active   44d
[kubectl] kube-system                  Active   44d
[kubectl] local-path-storage           Active   5d1h
[kubectl] olm                          Active   4d22h
Running kubectl with args [kubectl create configmap config-map-for-cleanup --from-literal type=testing --from-literal team=odo -n cmd-pref-config-test155xnv]
[kubectl] configmap/config-map-for-cleanup created
Current working dir: /home/wikus/syncthing/files/dev/go/src/odo/tests/integration
Running odo with args [odo registry update DefaultDevfileRegistry https://registry.stage.devfile.io -f]
[odo] Successfully updated registry
Running odo with args [odo create nodejs --project cmd-pref-config-test155xnv --context /tmp/2747527892 --devfile /home/wikus/syncthing/files/dev/go/src/odo/tests/examples/source/devfiles/nodejs/devfile-registry.yaml]
[odo] Devfile Object Creation
[odo]  •  Creating a devfile component from devfile path: /home/wikus/syncthing/files/dev/go/src/odo/tests/examples/source/devfiles/nodejs/devfile-registry.yaml  ...
[odo] 
 ✓  Creating a devfile component from devfile path: /home/wikus/syncthing/files/dev/go/src/odo/tests/examples/source/devfiles/nodejs/devfile-registry.yaml [130226ns]
[odo] Validation
[odo]  •  Validating if devfile name is correct  ...
[odo] 
 ✓  Validating if devfile name is correct [9321ns]
[odo]  •  Validating the devfile for odo  ...
[odo] 
 ✓  Validating the devfile for odo [3ms]
[odo]  •  Updating the devfile with component name "nodejs"  ...
[odo] 
 ✓  Updating the devfile with component name "nodejs" [427749ns]
[odo] 
[odo] Please use `odo push` command to create the component with source deployed
Running oc with args [oc whoami -t]
[oc] error: no token is currently in use for this session
Running oc with args [oc login --token ]
[oc] error: couldn't get https://192.168.1.92:6443/.well-known/oauth-authorization-server: unexpected response status 401
Deleting project: cmd-pref-config-test155xnv
Running kubectl with args [kubectl delete namespaces cmd-pref-config-test155xnv]
[kubectl] namespace "cmd-pref-config-test155xnv" deleted
Setting current dir to: /home/wikus/syncthing/files/dev/go/src/odo/tests/integration
Deleting dir: /tmp/2747527892
Deleting dir: /tmp/2270213049
[91m[1m• Failure in Spec Setup (BeforeEach) [6.411 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  when viewing local config without logging into the OpenShift cluster
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:141[0m
    [91m[1mwhen user is working with a devfile component [BeforeEach][0m
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:154[0m
      should set, list and delete config envs successfully
      [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:164[0m

      [91mNo future change is possible.  Bailing out early after 0.153s.
      Expected
          <int>: 1
      to match exit code:
          <int>: 0[0m

      /home/wikus/syncthing/files/dev/go/src/odo/tests/helper/helper_oc.go:214
[90m------------------------------[0m
Created dir: /tmp/2816930480
Created dir: /tmp/1816325923
Setting KUBECONFIG=/tmp/1816325923/config
Creating a new project: cmd-pref-config-test164cny
Running kubectl with args [kubectl create namespace cmd-pref-config-test164cny]
[kubectl] namespace/cmd-pref-config-test164cny created
Running kubectl with args [kubectl config set-context --current --namespace cmd-pref-config-test164cny]
[kubectl] Context "k0s-cluster" modified.
Running kubectl with args [kubectl get namespaces]
[kubectl] NAME                         STATUS   AGE
[kubectl] cert-manager                 Active   29d
[kubectl] cmd-pref-config-test164cny   Active   0s
[kubectl] default                      Active   44d
[kubectl] ingress-nginx                Active   41d
[kubectl] kube-node-lease              Active   44d
[kubectl] kube-public                  Active   44d
[kubectl] kube-system                  Active   44d
[kubectl] local-path-storage           Active   5d1h
[kubectl] olm                          Active   4d22h
Running kubectl with args [kubectl create configmap config-map-for-cleanup --from-literal type=testing --from-literal team=odo -n cmd-pref-config-test164cny]
[kubectl] configmap/config-map-for-cleanup created
Current working dir: /home/wikus/syncthing/files/dev/go/src/odo/tests/integration
Running odo with args [odo registry update DefaultDevfileRegistry https://registry.stage.devfile.io -f]
[odo] Successfully updated registry
Running odo with args [odo create nodejs --project cmd-pref-config-test164cny --context /tmp/2816930480 --devfile /home/wikus/syncthing/files/dev/go/src/odo/tests/examples/source/devfiles/nodejs/devfile-registry.yaml]
[odo] Devfile Object Creation
[odo]  •  Creating a devfile component from devfile path: /home/wikus/syncthing/files/dev/go/src/odo/tests/examples/source/devfiles/nodejs/devfile-registry.yaml  ...
[odo] 
 ✓  Creating a devfile component from devfile path: /home/wikus/syncthing/files/dev/go/src/odo/tests/examples/source/devfiles/nodejs/devfile-registry.yaml [185511ns]
[odo] Validation
[odo]  •  Validating if devfile name is correct  ...
[odo] 
 ✓  Validating if devfile name is correct [15565ns]
[odo]  •  Validating the devfile for odo  ...
[odo] 
 ✓  Validating the devfile for odo [5ms]
[odo]  •  Updating the devfile with component name "nodejs"  ...
[odo] 
 ✓  Updating the devfile with component name "nodejs" [450345ns]
[odo] 
[odo] Please use `odo push` command to create the component with source deployed
Running oc with args [oc whoami -t]
[oc] error: no token is currently in use for this session
Running oc with args [oc login --token ]
[oc] error: couldn't get https://192.168.1.92:6443/.well-known/oauth-authorization-server: unexpected response status 401
Deleting project: cmd-pref-config-test164cny
Running kubectl with args [kubectl delete namespaces cmd-pref-config-test164cny]
[kubectl] namespace "cmd-pref-config-test164cny" deleted
Setting current dir to: /home/wikus/syncthing/files/dev/go/src/odo/tests/integration
Deleting dir: /tmp/2816930480
Deleting dir: /tmp/1816325923
[32m• [SLOW TEST:6.258 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  When no ConsentTelemetry preference value is set
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:178[0m
    should not prompt when user calls for help
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:184[0m
[90m------------------------------[0m
[32m• [SLOW TEST:6.531 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  When no ConsentTelemetry preference value is set
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:178[0m
    should not prompt when preference command is run
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:189[0m
[90m------------------------------[0m
[32m• [SLOW TEST:6.059 seconds][0m
odo preference and config command tests
[90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:15[0m
  When ConsentTelemetry preference value is set
  [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:201[0m
    should not prompt the user
    [90m/home/wikus/syncthing/files/dev/go/src/odo/tests/integration/cmd_pref_config_test.go:203[0m
[90m------------------------------[0m
[36mS[0m[36mS[0m

[91m[1mSummarizing 2 Failures:[0m

[91m[1m[Fail] [0m[90modo preference and config command tests [0m[0mwhen viewing local config without logging into the OpenShift cluster [0m[91m[1m[BeforeEach] when user is working with a devfile component [0m[0mshould set, list and delete config successfully [0m
[37m/home/wikus/syncthing/files/dev/go/src/odo/tests/helper/helper_oc.go:214[0m

[91m[1m[Fail] [0m[90modo preference and config command tests [0m[0mwhen viewing local config without logging into the OpenShift cluster [0m[91m[1m[BeforeEach] when user is working with a devfile component [0m[0mshould set, list and delete config envs successfully [0m
[37m/home/wikus/syncthing/files/dev/go/src/odo/tests/helper/helper_oc.go:214[0m

[1m[91mRan 14 of 48 Specs in 97.608 seconds[0m
[1m[91mFAIL![0m -- [32m[1m12 Passed[0m | [91m[1m2 Failed[0m | [33m[1m0 Pending[0m | [36m[1m34 Skipped[0m
--- FAIL: TestIntegration (97.61s)
FAIL

Ginkgo ran 1 suite in 1m39.576598444s
Test Suite Failed
```

--- a/tests/integration/cmd_pref_config_test.go
+++ b/tests/integration/cmd_pref_config_test.go
@@ -142,6 +142,9 @@ var _ = Describe("odo preference and config command tests", func() {
 		var ocRunner helper.OcRunner
 		var token string
 		BeforeEach(func() {
+			if helper.IsKubernetesCluster() {
+				Skip("skipping for kubernetes until we can figure out how to simulate logged out state there")
+			}
 			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
 			helper.Cmd("odo", "create", "nodejs", "--project", commonVar.Project, "--context", commonVar.Context, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-registry.yaml")).ShouldPass()
 			ocRunner = helper.NewOcRunner("oc")
@@ -153,18 +156,12 @@ var _ = Describe("odo preference and config command tests", func() {
 		})
 		When("user is working with a devfile component", func() {
 			It("should set, list and delete config successfully", func() {
-				if helper.IsKubernetesCluster() {
-					Skip("skipping for kubernetes until we can figure out how to simulate logged out state there")
-				}
 				helper.Cmd("odo", "config", "set", "--force", "--context", commonVar.Context, "Name", "foobar").ShouldPass()
 				configValue := helper.Cmd("odo", "config", "view", "--context", commonVar.Context).ShouldPass().Out()
 				Expect(configValue).To(ContainSubstring("foobar"))
 				helper.Cmd("odo", "config", "unset", "--force", "--context", commonVar.Context, "Name").ShouldPass()
 			})
 			It("should set, list and delete config envs successfully", func() {
-				if helper.IsKubernetesCluster() {
-					Skip("skipping for kubernetes until we can figure out how to simulate logged out state there")
-				}
 				multiLineContent := `line1
 line2
 line3`

